### PR TITLE
Easier way to configure a pool

### LINKED ISSUE #432: Provide easier way to configure a pool with a simple set of endpoints
Weighted ones are fun, but also irrelevant in many cases. Simple things should be done easily, so some more convenient way to pass a set of endpoints to the pool is needed.

### PR DESCRIPTION
**Pull request by smallhive**
close #432

### CODE REVIEW

File: pool/pool.go [hunk 1]
```diff
@@ -1406,6 +1406,48 @@ const (
...
+	return params
+}
+
+// New creates connection pool using simple set of endpoints and parameters.
+//
+// See also [pool.DefaultOptions] for details.
+//
+// Returned errors:
+//   - [neofscrypto.ErrIncorrectSigner]
+func New(endpoints []string, options InitParameters) (*Pool, error) {
```
Comment (roman-khimov) [inline pool/pool.go:1431]: How about
`New(endpoints []NodeParam, signer neofscrypto.Signer, options InitParameters) (*Pool, error)`
and
`NewFlatNodeParams(endpoints []string) []NodeParam`

This way we can replace `NewPool` completely. It's a little less convenient, but more versatile at the same time.

Comment (smallhive) [reply pool/pool.go]: Looks not so easy, but makes sense. Updated

Comment (roman-khimov) [reply pool/pool.go]: We can make them two methods, opinions are welcome, but I think one is enough.

File: pool/pool.go [hunk 2]
```diff
@@ -1406,6 +1406,58 @@ const (
...
+		nodeDialTimeout:           defaultDialTimeout,
+		nodeStreamTimeout:         defaultStreamTimeout,
+	}
+
+	return params
+}
+
+// New creates connection pool using simple set of endpoints and parameters.
+//
+// See also [pool.DefaultOptions] for details.
```
[the remark this answers is not in the document]
Comment (smallhive) [reply pool/pool.go]: Updated

**Review by carpawell — APPROVED** [review]

**Review by cthulhu-rider — APPROVED** [review]

**Pull request merged**